src/main/java/models/task/TaskList.java

```diff
@@ -147,46 +147,117 @@ public int getSize() {
      * @param taskIndexNumber Index of task to be edited.
      * @param updatedTaskRequirements Array containing indexes of task requirements to be removed and
      *                                task requirements to be added.
+     * @return String array of success and error messages to be printed.
      */
-    public void editTaskRequirements(int taskIndexNumber, String updatedTaskRequirements) {
+    public String[] editTaskRequirements(int taskIndexNumber, String updatedTaskRequirements) {
+        ArrayList<String> successMessages = new ArrayList<>();
+        successMessages.add("Success!");
+        ArrayList<String> errorMessages = new ArrayList<>();
+        errorMessages.add("Errors...");
+
         ArrayList<String> newTaskRequirementDetails = parserHelper.parseTaskRequirementDetails(updatedTaskRequirements);
+
+        if ("No error".equals(newTaskRequirementDetails.get(0))) {
+            newTaskRequirementDetails.remove(0);
+        } else {
+            while (!"Errors end".equals(newTaskRequirementDetails.get(0))) {
+                errorMessages.add(newTaskRequirementDetails.get(0));
```

Comment:
Not sure about your implementation here, but ParserHelper itself has an ArrayList<String> that can help you store any error messages in parsing. You can simply call ParserHelper.getErrorMessages() to get all the messages, so that could possibly save you some work here?

Reply:
it's because i'm returning a single String array back so i use that as an indicator to separate the error messages and the new task requirement details

Reply:
Oops ok I see that you are already using the errorMessages array! Sorry about this!

Reply:
oh damn i didn't know, thanks i'll go change it